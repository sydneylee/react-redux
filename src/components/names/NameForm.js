import React, {Component} from 'react';
import PropTypes from 'prop-types';

class NameForm extends Component {
    // static defaultProps = {
    //     name : 'default name',
    //     onChange : ()=>console.warn('TODO : onChange'),
    //     onSubmit : ()=>console.warn('TODO : onSubmit'),
    // };

    static propTypes = {};

    //state = {};

    handleOnChange=(e)=>{
        console.log(e.target.value);
        let payload = {};
        payload[e.target.name]=e.target.value;
        this.props.onChange(payload);
        // if(e.target.name == 'firstname') {
        //     this.props.onChange(e.target.value);
        // }else{
        //
        // }
    };

    //TODO : onSubmit에서 e를 이용해서 form전체의 값을 받기
    nameRef=null;
    handleOnSubmit=(e)=>{
        e.preventDefault();
        let data = new FormData(e.target);
        // console.log('data');
        // for(var pair of data.entries()) {
        //     console.log(pair[0]+', '+pair[0]);
        // }
        let payload = {};
        for(var pair of data.entries()) {
            //console.log(pair[0]+', '+pair[0]);
            payload[pair[0]]=pair[1];
        }
        this.props.onSubmit(payload);
        //this.props.onSubmit(this.nameRef.value);
    };



    render() {
        const {fullname} = this.props;
        return (
            <form onSubmit={this.handleOnSubmit}>
                <input type="text"
                       name='firstname'
                       value={fullname.firstname}
                       onChange={this.handleOnChange}

                />
                <input type="text"
                       name='lastname'
                       value={fullname.lastname}
                       onChange={this.handleOnChange}

                />
                <button type="submit">submit</button>
            </form>
        );
    }
}

export default NameForm;
