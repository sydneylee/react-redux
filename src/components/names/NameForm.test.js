import React from 'react';
import { shallow } from 'enzyme';
import NameForm from './NameForm';

describe('NameForm', () => {
    let component = null;

    it('renders correctly', () => {
        component = shallow(<NameForm />);
    });

    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});