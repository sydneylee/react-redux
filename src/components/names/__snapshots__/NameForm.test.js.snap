// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameForm matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="firstname"
    onChange={[Function]}
    placeholder="first name"
    type="text"
    value=""
  />
  <input
    name="lastname"
    onChange={[Function]}
    placeholder="last name"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    submit
  </button>
  <div
    style={
      Object {
        "fontSize": "10px",
      }
    }
  />
</form>
`;
